### PR TITLE
Edit cohort

### PR DESCRIPTION
**Pull request by lubnaabd**

### CODE REVIEW

File: public/js/editCohort.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+const editCohort = document.getElementById('editCohort');
+const names = document.getElementById('name');
+const descriptions = document.getElementById('description');
+const githublinks = document.getElementById('githublink');
+const imgURls = document.getElementById('imgURl');
+const nameerror = document.getElementById('nameerror');
```
Comment (NouraldinS) [inline public/js/editCohort.js]: Camelcase those.

File: public/js/editCohort.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+editCohort.addEventListener('click', (e) => {
+  if (checkName() && checkDescriptions() && checkGithublink() && checkImage()) {
+    const URL = window.location.href;
+    const splitUrl = URL.split('/');
+    const ID = splitUrl[splitUrl.length - 1];
+    const name = names.value;
+    const description = descriptions.value;
+    const githublink = githublinks.value;
+    const imgURl = imgURls.value;
```
Comment (NouraldinS) [inline public/js/editCohort.js]: So you've written what's above all in lowercase, and here you say `imgURl`? Fix your naming cases.

File: public/js/editCohort.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    const cohortID = JSON.parse(ID);
```
Comment (NouraldinS) [inline public/js/editCohort.js]: `id` not `ID`.

File: public/js/editCohort.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    const data = {
+      name, description, githublink, imgURl,
+    };
+    fetch(`/admin/cohorts/edit/${cohortID}`, {
+      method: 'POST',
+      credentials: 'same-origin',
+      headers: { 'Content-Type': 'application/json; charset=utf-8' },
+      body: JSON.stringify(data),
+    })
+      .then((response) => { response.json(); })
```
Comment (NouraldinS) [inline public/js/editCohort.js]: either
```js
.then((response) => response.json())
```
or
```js
.then((response) => {
  return response.json();
}
```
Don't mix up both.

File: src/controllers/editcohort.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+
+exports.editCohort = (req, response, next) => {
+  const data = req.body;
+  const cohortID = JSON.parse(req.params.id);
```
Comment (NouraldinS) [inline src/controllers/editcohort.js]: Destructure id before using it here, and rename your `cohortID` to `cohortId`.

File: src/database/queries/editcohort.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+const dbconnection = require('../db_connection');
+
+const editCohorts = (name, description, githublink, imgURl, cohortID) => new Promise((resolve, reject) => {
```
Comment (NouraldinS) [inline src/database/queries/editcohort.js]: Don't pass all of these as parameters, pass them to this function as one object and destructure them inside the `editCohorts`.
i.e.
```js
// Callee file
editCohorts({ name, description, githubLink, imgUrl, cohortId });

// editCohort.js
const editCohorts = (cohortData) => {
  const { name, description, githublink, imgURl, cohortID } = cohortData;
...
```
Too, rename your `githublink` to `githubLink`, your `imgURl` to `imgUrl` or `iMgUrL`, your `cohortID` to `cohortId`, and your `editcohort.js` to `editCohorts.js`

File: src/views/cohort.hbs [hunk 7]
```diff
@@ -25,7 +25,7 @@
           <a href="{{imgurl}}">click</a>
         </td>
         <td>
-          <a>
+          <a href="/admin/cohorts/edit/{{this.id}}" id="{{this.id}}">
             Edit </a>
```
Comment (NouraldinS) [inline src/views/cohort.hbs]: Fix indentation.

File: src/views/editcohort.hbs [hunk 8]
```diff
@@ -0,0 +1,30 @@
+<section class="content">
+        {{#each results}}
```
Comment (NouraldinS) [inline src/views/editcohort.hbs]: Fix indentation .

**Review by NouraldinS — CHANGES_REQUESTED** [review]

**Review by HemaSAli — APPROVED** [review]

**Review by ali-7 — APPROVED** [review]

File: src/controllers/editcohort.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+
```
Comment (NouraldinS) [inline src/controllers/editcohort.js]: Rename file to `editCohort.js`.

File: src/database/queries/editcohort.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
+const dbconnection = require('../db_connection');
```
Comment (NouraldinS) [inline src/database/queries/editcohort.js]: Delete unused file.

**Review by NouraldinS — CHANGES_REQUESTED** [review]

**Review by NouraldinS — APPROVED** [review]

**Pull request merged**